<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>608</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hough变换</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="show_img">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>640</width>
      <height>480</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>640</width>
      <height>480</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>640</width>
      <height>480</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="canny">
    <property name="geometry">
     <rect>
      <x>655</x>
      <y>90</y>
      <width>131</width>
      <height>401</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Canny</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="3" column="0">
      <widget class="QSpinBox" name="canny_spinBox_th1"/>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="canny_spinBox_th2"/>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QSpinBox" name="Max_canny_spinBox"/>
     </item>
     <item row="2" column="0">
      <widget class="QSlider" name="canny_Slider_th1">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>th2</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSlider" name="canny_Slider_th2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Max Value</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>th1</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>510</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>WangHongshuo</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>0</y>
      <width>121</width>
      <height>81</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0">
      <widget class="QPushButton" name="open_file">
       <property name="text">
        <string>打开图片</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QCheckBox" name="hough_checkBox">
       <property name="text">
        <string>Hough</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QRadioButton" name="circles">
       <property name="text">
        <string>圆</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QRadioButton" name="lines">
       <property name="text">
        <string>线</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="canny_checkBox">
       <property name="text">
        <string>Canny</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QCheckBox" name="blur">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>50</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>高斯模糊</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>501</y>
      <width>681</width>
      <height>50</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="lines_color">
       <property name="title">
        <string>线色</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="1">
         <widget class="QRadioButton" name="lines_color_blue">
          <property name="text">
           <string>蓝色</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QRadioButton" name="lines_color_green">
          <property name="text">
           <string>绿色</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QRadioButton" name="lines_color_white">
          <property name="text">
           <string>白色</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QRadioButton" name="lines_color_black">
          <property name="text">
           <string>黑色</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QRadioButton" name="lines_color_red">
          <property name="text">
           <string>红色</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="hough">
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <widget class="QSpinBox" name="line_lenth_spinBox"/>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Hough阈值</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QSlider" name="line_lenth_Slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="test">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>530</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>test</string>
    </property>
   </widget>
   <zorder>show_img</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>canny</zorder>
   <zorder>label_5</zorder>
   <zorder>blur</zorder>
   <zorder>test</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>803</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionopen"/>
    <addaction name="actionexit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionabot"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionopen">
   <property name="text">
    <string>打开图片</string>
   </property>
  </action>
  <action name="actionexit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="actionabot">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>canny_Slider_th1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canny_spinBox_th1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>683</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>692</x>
     <y>452</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canny_Slider_th2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canny_spinBox_th2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>746</x>
     <y>452</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canny_spinBox_th2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canny_Slider_th2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>746</x>
     <y>452</y>
    </hint>
    <hint type="destinationlabel">
     <x>737</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canny_spinBox_th1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>canny_Slider_th1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>692</x>
     <y>452</y>
    </hint>
    <hint type="destinationlabel">
     <x>683</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionexit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
